src/main/java/ch/njol/skript/config/Node.java

```diff
@@ -114,37 +115,113 @@ public void move(final SectionNode newParent) {
 		p.remove(this);
 		newParent.add(this);
 	}
-	
-	@SuppressWarnings("null")
-	private final static Pattern linePattern = Pattern.compile("^((?:[^#]|##)*)(\\s*#(?!#).*)$");
-	
+
 	/**
 	 * Splits a line into value and comment.
 	 * <p>
-	 * Whitespace is preserved (whitespace in front of the comment is added to the value), and any ## in the value are replaced by a single #. The comment is returned with a
+	 * Whitespace is preserved (whitespace in front of the comment is added to the value), and any ## not in quoted strings in the value are replaced by a single #. The comment is returned with a
 	 * leading #, except if there is no comment in which case it will be the empty string.
 	 * 
 	 * @param line
 	 * @return A pair (value, comment).
 	 */
-	public static NonNullPair<String, String> splitLine(final String line) {
+	public static NonNullPair<String, String> splitLine(String line) {
```

Comment:
This has conflicts with feature that I can't resolve for you, because something else also changed this method. You'll probably need to work out how to integrate the two.